src/main/java/com/linkedin/kmf/services/StatsdMetricsReporterService.java

```diff
@@ -47,17 +47,13 @@ public StatsdMetricsReporterService(Map<String, Object> props, String name) {
 
   @Override
   public synchronized void start() {
-    _executor.scheduleAtFixedRate(
-      new Runnable() {
-        @Override
-        public void run() {
-          try {
-            reportMetrics();
-          } catch (Exception e) {
-            LOG.error(_name + "/StatsdMetricsReporterService failed to report metrics", e);
-          }
-        }
-      }, _reportIntervalSec, _reportIntervalSec, TimeUnit.SECONDS
+    _executor.scheduleAtFixedRate(() -> {
```

Comment:
I noticed that usage of the executor service is kind of prevalent in kafka-monitor. Instead of using the `ScheduledExecutorService` explicitly, would it be a good idea to write a thin wrapper/abstraction which takes in an executor and provided an unified API/method to schedule periodical task ? For example

```java
  class KmfExecutor {
    // ...
    public KmfExecutor(ScheduledExecutorService executor) {
      // ...
    }

    public void scheduleAtFixedRate(/* some function */, long initialDelay, long period, String errorMsg) {
      // ...
    }

    public void awaitTermination() {
      // ...
    }

    public void shutdown() {
      // ...
    }
    // ...
  }
```

Then instantiate them all together and provide each component that needs executor with this `KmfExecutor` so that each component do not have to create its own ad-hoc executor. Here are a few advantages:

1. Increase testability. We can potentially mock the `KmfExecutor` or make the `KmfExecutor` takes in mocked executor.

2. We can potentially have a centralized class which contains all executors and add metrics to this "executor manager"

It is just my thoughts and just want to let you know in case you think this is a good idea or potentially something you might want to do. You don't have to make any change at this point.

Reply:
Composing a `wrapper or abstraction` which takes in an executor and provides a **unified API** and method to _schedule periodical task_ sounds like a reasonable idea.

I am also for avoiding individually-owned ad-hoc executor. I can do this in the future in spare time for the benefits of increased `testability` and `class centralization`.